dnd5esheets/front/src/components/CharacterSheet.tsx

```diff
@@ -214,12 +214,15 @@ export default function CharacterSheet(props: {
                       modifier={
                         props.character.data.abilities[ability].modifier
                       }
-                      onChange={(score: number) =>
+                      onChange={(score: number) => {
+                        if (Number.isNaN(score)) {
+                          score = 0
```

Comment:
We tend to avoid mutations altogether, but especially of the parameters.
Instead I'd write that:
```typescript
character.data.abilities[ability].score = Number. IsNaN(score) ? 0 : score
```

Also, this sanitization feels like it belongs to the `ScoreBox` component rat
her than here, and looking there, I realize I tried to implement it, but failed 😅

Would you be ok moving your validation there?

Reply:
I did that, but _think_ it's actually better to have `character.data.abilities[ability].score = Number. IsNaN(score) ? 0 : score` in `CharacterSheet`. This way, we don't have `NaN` as an input of the `ScoreBox`, which is then taken at face value for the computation of other scores.

<img width="358" alt="Screenshot 2023-08-12 at 11 24 37" src="https://github.com/brouberol/5esheets/assets/480131/af99f3e7-9790-4703-ad2d-14b850000c47">

WDYT?

Reply:
Either way we don't have NaN ending up in the character sheet, do we?
And validating it within `ScoreBox` means the NaN won't leave the scope of the input component, which we could expect to always yield a number.
(Even though, interestingly enough, in typescript `NaN` is actually a `number` [apparently](https://www.typescriptlang.org/play?#code/FAYw9gdgzgLgBAQwFxwgVwLYCMCmAnOAXjgDkESg))

Reply:
We do actually. If we don't filter out `NaN` from the `CharacterSheet` and only format it as `0` in the `ScoreBox`, other components use the `NaN` value to compute their own value (passive perception value, ability and skill modifiers), which ends up being `NaN` as well. Cf the previous screenshot.

If we filter out `NaN` value at the source (ie the `CharacterSheet`), we don't have such a propagation of `NaN` values.

Reply:
I can't make sense of it, and unable to check on a computer. Let's merge that and I'll have a look later 🙂